src/validation.h

```diff
@@ -475,9 +475,19 @@ class BlockManager {
      */
     std::multimap<CBlockIndex*, CBlockIndex*> m_blocks_unlinked;
 
+    /**
+     * Load the blocktree off disk and into memory. Populate certain metadata
+     * per index entry (nStatus, nChainWork, nTimeMax, etc.) as well as peripheral
+     * collections like setDirtyBlockIndex.
```

Comment:
hmmm maybe setDirtyBlockIndex could be part of BlockManager in future commits given this set of block index  is also shared between chainstates? And would let implement batching managed by BlockManager